services/storage/src/simcore_service_storage/modules/celery/worker.py

```diff
@@ -0,0 +1,32 @@
+import logging
+from collections.abc import Callable
+
+from celery import Celery
+from celery.contrib.abortable import AbortableTask
+from models_library.progress_bar import ProgressReport
+
+from .models import TaskID
+
+_logger = logging.getLogger(__name__)
+
+
+class CeleryTaskQueueWorker:
+    def __init__(self, celery_app: Celery) -> None:
+        self.celery_app = celery_app
+
+    def register_task(self, fn: Callable, task_name: str | None = None) -> None:
+        name = task_name or fn.__name__
+        _logger.info("Registering %s task", name)
+        self.celery_app.task(name=name, base=AbortableTask, bind=True)(fn)
```

Comment:
I find shame that you are missing the decorator here ...
IMO there should be two versions of `register_task`.

```python
worker.add_task(fun) # instead of register_task

@worker.as_task()
def other_fun():
     ...
```

```python
    def as_task(self, task_name: str | None = None) -> Callable:
        def decorator(fn: Callable) -> Task:
            task = self.add_task(fn, task_name)
            return task
        return decorator  
```